### PR TITLE
Concatenation of multiple simple operations and negative numbers

### PR DESCRIPTION
**Pull request by brenpardal**
### **WHAT'S NEW**
**This update includes:**
*  Concatenation of multiple simple operations
*  Implementation of negative numbers in operations

edge_cases

### CODE REVIEW

File: app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt [hunk 1]
```diff
@@ -45,15 +42,28 @@ class CalculatorModel() : CalculatorContract.Model {
...
     override fun setOperationSymbolPressed(operatorSymbol: String) {
-        if (!firstOperand.isEmpty()) {
+        if (firstOperand.isEmpty() && operatorSymbol == OPERATOR_SUB) {
+            firstOperand += operatorSymbol
+        } else if (secondOperand.isEmpty() && !operator.isEmpty() && operatorSymbol == OPERATOR_SUB) {
+            secondOperand += operatorSymbol
+        } else if (!firstOperand.isEmpty()) {
             operator = operatorSymbol
-            setPartial()
         }
```
Comment (nicolbidone) [inline app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt:51]: instead of nested if-else, here you can use when{ (without argument)

Comment (brenpardal) [reply app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt:49]: Solved. Thanks!!

**Review by BQuinteros — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

File: app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt [hunk 2]
```diff
@@ -19,7 +19,7 @@ class CalculatorModel() : CalculatorContract.Model {
     private var equalsPressed = false
     private var res = EMPTY_STRING
     val decimalFormat = DecimalFormat(DECIMAL_FORMAT)
-    override var errorEnum = ErrorEnum.NONE
+    override var errorEnum = ResultEnum.SUCCESS
```
Comment (SartiMau) [inline app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt:22]: same here

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/calculatorkt/mvp/contract/CalculatorContract.kt [hunk 3]
```diff
@@ -1,11 +1,11 @@
 package com.calculatorkt.mvp.contract
 
-import com.calculatorkt.utils.ErrorEnum
+import com.calculatorkt.utils.ResultEnum
 
 interface CalculatorContract {
 
     interface Model {
-        var errorEnum: ErrorEnum
+        var errorEnum: ResultEnum
```
[the remark this answers is not in the document]
Comment (brenpardal) [reply app/src/main/java/com/calculatorkt/mvp/contract/CalculatorContract.kt]: Solved. Replaced errorEnum with resultEnum.

File: app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt (hunk 2 above)
Comment (brenpardal) [reply app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt]: Solved. Replaced errorEnum with resultEnum.

**Review by SartiMau — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**